[Hero] Design guidelines (#304)
primer/brand: apps/docs/content/components/Hero/index.mdx

---
title: Hero

> description: Use the hero component to render a full-width banner on a webpage

Reviewer: Suggestion around making 'top of the page' use case clear upfront

```suggestion
Use the hero component to render a full-width banner at the top of a page
```

---

import CustomVideoPlayer from '../../../src/components/custom-video-player'
import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

## Anatomy

![An image displaying a hero with a heading, description, primary and secondary actions.](https://github.com/primer/brand/assets/912236/51f3dd23-9ac7-4825-9169-4fb7f295e7f1)

- Heading: the main title of the hero.
- Description: a short description of the hero.
- Actions: the main and secondary actions the user should take on the page.

## Usage

> Use the hero component to provide a title, description and actions to introduce the main content of a page. The hero component is designed to be positioned at the top of a page and spans the full width of the viewport.

Reviewer: Total nit.. a grammatical suggestion to use a relative clause that reduce repetition of 'to'...

```suggestion
Use the hero component to provide a title, description and action that introduces the main content of a page. The hero component is designed to be positioned at the top of a page and spans the full width of the viewport.
```

Both heading and description of the hero should be short and descriptive. Heading should be not be longer than 2 lines, and description should be no longer than 3-4 lines.

> The hero component is not designed to be used as a banner or advertisement, or to provide a lot of information to the user. Consider using other components, such as the [section intro](/components/SectionIntro) or the [river](/components/River) instead.

Reviewer: > used as a banner

Could we be more specific about this? Only because the description in the front-matter describes this component as a `full-width banner`, so clarity may be helpful on which type of banner we don't want to qualify this as.

Reviewer: Also worth adding CTA Banner into the mix?

```suggestion
The hero component is not designed to be used as a banner or advertisement, or to provide a lot of information to the user. Consider using other components, such as the [CTA banner](/components/CTABanner), [section intro](/components/SectionIntro) or the [river](/components/River) instead.
```

Reviewer 2: Agree!

Author: The `CTA banner` is used in most cases to repeat the content and actions the `Hero` introduces.

For this, I'd extend the first paragraph and suggest the `CTA Banner` as an optional component when not positioned in the top.

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/e1792455-f008-45fd-9dd6-7330bc58c03c" />
    <Caption>
      Use the hero component to introduce the page with short, descriptive text.
    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/4f60189c-c5d0-478b-bcb0-823bb1b5ee6f" />
    <Caption>
      Don't use long, complex text in the hero. Consider splitting the content
      using other components.
    </Caption>
  </Dont>
</DoDontContainer>

## Options

### Alignment

Hero elements can be aligned to the start or center of the page. By default, the hero is aligned to the start of the page to allow for an image or a visual asset to be positioned on the right of the hero. Use the center alignment if the asset is centered or not present, or the scenario calls for it.

<DoDontContainer stacked>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/3d96284d-46ae-4211-a291-9750ec54bbf6" />
    <Caption>
      Use left alignment if the hero includes an image or visual asset on the
      right. 
    </Caption>
  </Do>
  <Dont>

>     <img src="https://github.com/primer/brand/assets/912236/1f56d40e-c3aa-4955-aa8f-a4748e9fe149" />

Reviewer: Great example!

    <Caption>

>       Don't use center alignment if the hero includes a centered image or visual asset.

Reviewer 2: ```suggestion
      Don't use center alignment if the hero includes a right aligned image or visual asset.
```

Is this what you mean?

Author: I think the description is wrong in the sense that the image can be centered, left or right aligned and create visual noise combined with the text. I'll rephrase with something more generic 👍 

<img width="906" alt="Screenshot 2023-07-04 at 17 12 15" src="https://github.com/primer/brand/assets/912236/dde2f50e-c69a-4733-8a5f-3b1621dcb874">

The hero heading is rendered as an `h1` element by default. You can modify the heading level but be aware that this will affect the page's heading structure and may impact accessibility. [See heading accesibility](/components/Heading#accessibility) for more information.

The hero can optionally include primary and secondary actions. These actions are rendered as a [button group](/components/ButtonGroup) and follow the same guidelines as the button group component.

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/e1792455-f008-45fd-9dd6-7330bc58c03c" />
    <Caption>
      Use the hero component to introduce the page with short, descriptive text.
    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/4f60189c-c5d0-478b-bcb0-823bb1b5ee6f" />
    <Caption>
      Don't use long, complex text in the hero. Consider splitting the content
      using other components.
    </Caption>
  </Dont>
</DoDontContainer>

## Options

### Alignment

> Hero elements can be aligned to the start or center of the page. By default, the hero is aligned to the start of the page to allow for an image or a visual asset to be positioned on the left of the hero. Use the center alignment if the asset is centered or not present, or the scenario calls for it.

Reviewer 2: ```suggestion
Hero elements can be aligned to the start or center of the page. By default, the hero is aligned to the start of the page to allow for an image or a visual asset to be positioned on the right of the hero. Use the center alignment if the asset is centered or not present, or the scenario calls for it.
```

Reviewer 2: I think this is supposed to say the image is on the **right** and the text is on the **left**, but please correct me if I'm mistaken.

Author: You're right, thank you!

<DoDontContainer stacked>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/3d96284d-46ae-4211-a291-9750ec54bbf6" />
    <Caption>
      Use left alignment if the hero includes an image or visual asset on the

>       left.

Reviewer 2: ```suggestion
      right. 
```

Author: good catch 🙈

    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/1f56d40e-c3aa-4955-aa8f-a4748e9fe149" />
    <Caption>
      Don't use center alignment if the hero includes an image or visual asset

>       placed on the left.

Reviewer 2: Depending on the correction of the comment/suggestion above, this might have to change as it was a little confusing for me to read.
